applications/callflow/src/module/cf_dynamic_cid.erl

```diff
@@ -43,9 +48,104 @@
+            lager:info("user is choosing a caller id for this call from couchdb doc"),
+	    handle_list(Data, Call);
+        _ ->
+	    lager:info("user must manually enter on keypad the caller id for this call"),
+	    handle_manual(Data, Call)
+    end.
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Entry point for this module
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_list(wh_json:object(), whapps_call:call()) -> 'ok'.
+handle_list(Data, Call) ->
+    CallerIdNumber = whapps_call:caller_id_number(Call),
+    lager:debug("callerid number before this module: ~s ", [CallerIdNumber]),
+
+    {NewCidInfo, Dest} = get_list_entry(Data, Call),
+
+    NewCallerIdNumber = wh_json:get_value(<<"number">>, NewCidInfo),
+    NewCallerIdName = wh_json:get_value(<<"name">>, NewCidInfo),
+
+    lager:info("setting the caller id number to ~s", [NewCallerIdNumber]),
+
+    Updates = [{fun whapps_call:kvs_store/3, 'dynamic_cid', NewCallerIdNumber}
+	      ,{fun whapps_call:set_caller_id_number/2, NewCallerIdNumber}
+	      ,{fun whapps_call:set_caller_id_name/2, NewCallerIdName}
+              ],
+
+    cf_exe:set_call(whapps_call:exec(Updates, Call)),
+
+    lager:debug("destination number from cf_capture_group regex: ~s ", [Dest]),
+    Number = wnm_util:to_e164(Dest),
+    lager:info("send the call onto real destination of: ~s", [Number]),
+
+    maybe_route_to_callflow(Data, Call, Number).
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec maybe_restrict_call(wh_json:object(), whapps_call:call(), ne_binary(), wh_json:object()) -> 'ok'.
+maybe_restrict_call(Data, Call, Number, Flow) ->
+    case should_restrict_call(Data, Call, Number) of
+        'true' ->
+            lager:info("Not allowed to call this destination, terminate", []),
+            _ = whapps_call_command:answer(Call),
+            _ = whapps_call_command:prompt(<<"cf-unauthorized_call">>, Call),
+            _ = whapps_call_command:queued_hangup(Call),
+            'ok';
+        'false' ->
+            cf_exe:branch(wh_json:get_value(<<"flow">>, Flow), Call)
+    end.
+
+-spec maybe_route_to_callflow(wh_json:object(), whapps_call:call(), ne_binary()) -> 'ok'.
+maybe_route_to_callflow(Data, Call, Number) ->
+    case cf_util:lookup_callflow(Number, whapps_call:account_id(Call)) of
+        {'ok', Flow, NoMatch} ->
+            lager:info("callflow ~s satisfies request", [wh_json:get_value(<<"_id">>, Flow)]),
+            Updates = [{fun whapps_call:set_request/2
+                        ,list_to_binary([Number, "@", whapps_call:request_realm(Call)])
+                       }
+                       ,{fun whapps_call:set_to/2, list_to_binary([Number, "@", whapps_call:to_realm(Call)])}
+                       ,fun(C) when NoMatch ->
```

Comment:
@stormqloud 
Howdy,
Looks like you want to crash if the callflow found is not a NoMatch.  However, if not flow is found then you play a message to the caller. 

May I suggest that you update line 127 to:
{'ok', Flow, 'true'}
and remove this fun?

That way both conditions that this function filters for result in the same user experience?

Additionally, I would request some return to cf_exe after the prompt play for "disa-invalid_extension".  For example:
cf_exe:stop(Call).